openms-wiki/developer-faq.md

```diff
@@ -0,0 +1,491 @@
+
+The following options are of interest:
+
+* `CMAKE_BUILD_TYPE` Define if you want to build Debug or Release version of OpenMS. Release is the default.
+
+* `CMAKE_FIND_ROOT_PATH` The path to the `contrib` libraries.
+  > **_NOTE:_**  that you can also provide more then one value here (e.g., `-D CMAKE_FIND_ROOT_PATH="/path/to/contrib;/usr/"` will search in your `contrib` path and in `/usr` for the required libraries)
+
+* `STL_DEBUG` Enables STL debug mode.
+
+* `DB_TEST` (deprecated) Enables database testing.
+
+* `QT_DB_PLUGIN` (deprecated) Defines the db plugin used by Qt.
+
+* `MT_CUDA_BUILD_TYPE` ...
+
+View the description for each option by calling `ccmake`.
+
+### Can I use another solver other than GLPK?
+
+Other solvers can be used, but by default, the build system only links to GLPK (this is how OpenMS binary packages must be built).
+To to use another solver, use `cmake ... -D USE_COINOR=1 ....` and refer to the documentation of the `LPWrapper` class.
+
+### How do I switch to debug or release configuration?
+
+For Makefile generators (typically on Linux), you can set the `CMAKE_BUILD_TYPE` variable to either Debug or Release by calling `ccmake`.
+For Visual Studio, this is not necessary as all configurations are generated and you can choose the one you like within the IDE itself.
+The 'Debug' configuration enabled debug information. The 'Release' configuration disables debug information and enables optimisation.
+
+### I changed the `contrib` path, but re-running `CMake` won't change the library paths?
+
+Once a library is found and its location is stored in a cache variable, it will only be searched again if the corresponding entry in the cache file is set to false.
+You can delete the `CMakeCache.txt`, but all other custom settings will be lost.
+
+### What are the most useful `make` targets?
+
+In Visual Studio you can see all targets on the left. For Makefiles type make help. However, this list is quite long.
+The most useful targets will be shown to you by calling the targets target, i.e. make targets.
+
+### `CMake` can't seem to find a `Qt` library (usually `QtCore`). What now?
+
+`CMake` finds `QT` by looking for `qmake` in your PATH or for the Environment Variable `QTDIR`. Set these accordingly.
+If the problem still persists: do you have a second installation of Qt (especially the MinGW version)? This might lead ``CMake`` to the wrong path (it's searching for the ``Qt*.lib`` files).
+You should only move or delete the offending `Qt` version if you know what you are doing!
+A save workaround is to edit the `CMakeCache` file (e.g. via `ccmake`) and set all paths relating to `QT` (e.g. `QT_LIBRARY_DIR`) manually.
+
+### (Windows) What version of Visual Studio should I use?
+
+Use the latest if possible. Get the latest `CMake`, as its generator needs to support your VS. If your VS is too new and there is no `CMake` for that yet, you're gonna be faced with a lot of conversion issues.
```

Comment:
> Use the latest if possible.

It is recommended to use the latest version.

Reply:
> If your VS is too new and there is no `CMake` for that yet, you're gonna be faced with a lot of conversion issues.

If VS is too new and there is no `CMake` for that yet, there might be a lot of conversion issues.